src/traccuracy/metrics/_divisions.py

```diff
@@ -143,3 +140,31 @@ def _calculate_metrics(self, g_gt: TrackingGraph, g_pred: TrackingGraph) -> dict
             "False Negative Divisions": fn_division_count,
             "Wrong Children Divisions": wc_division_count,
         }
+        for fb in range(1, self.frame_buffer + 1):
+            new_tp_div_count = 0
+            for node in g_pred.graph.nodes:
```

Comment:
Can this loop be shortened by just iterating over the division nodes?

Reply:
This is a good question, and I think it's worth us discussing in other contexts. I'm not sure. `TrackingGraph.get_divisions()` technically runs a loop of length `number_of_nodes`, because it gets the `out_degree` and then filters the nodes by degree. So it could actually be slower to `get_divisions` + iterate through divisions than to just iterate through all nodes?

Reply:
I think in this case it would probably still be quicker, since we won't be getting the node info for unrelated nodes, so I've updated it. I think it would be a good idea to consider more broadly whether some of our convenience functions are inadvertently increasing complexity. Maybe we should store dividing nodes on graph build so that we can access them without iterating?